RecoTracker/MkFitCore/interface/HitStructures.h

```diff
@@ -0,0 +1,874 @@
+
+    void Reset() {
+      for (auto& i : m_layers_of_hits) {
+        i.Reset();
+      }
+    }
+
+    void SuckInHits(int layer, const HitVec& hitv) {
+      m_layers_of_hits[layer].SuckInHits(hitv);
+      /*
+    int   nh  = hitv.size();
+    auto &loh = m_layers_of_hits[layer];
+    loh.BeginRegistrationOfHits(hitv);
+    for (int i = 0; i < nh; ++i) loh.RegisterHit(i);
+    loh.EndRegistrationOfHits();
+    */
+    }
+
+    void SuckInDeads(int layer, const DeadVec& deadv) { m_layers_of_hits[layer].SuckInDeads(deadv); }
+
+    void SetBeamSpot(const BeamSpot& bs) { m_beam_spot = bs; }
+
+    LayerOfHits& operator[](int i) { return m_layers_of_hits[i]; }
+    const LayerOfHits& operator[](int i) const { return m_layers_of_hits[i]; }
+  };
+
+  //==============================================================================
+  // TrackCand, CombinedCandidate and EventOfCombinedCandidates
+  //==============================================================================
+
+  struct HoTNode {
+    HitOnTrack m_hot;
+    float m_chi2;
+    int m_prev_idx;
+  };
+
+  struct HitMatch {
+    int m_hit_idx = -1;
+    int m_module_id = -1;
+    float m_chi2 = 1e9;
+
+    void reset() {
+      m_hit_idx = -1;
+      m_module_id = -1;
+      m_chi2 = 1e9;
+    }
+  };
+
+  struct HitMatchPair {
+    HitMatch M[2];
+
+    void reset() {
+      M[0].reset();
+      M[1].reset();
+    }
+
+    void consider_hit_for_overlap(int hit_idx, int module_id, float chi2) {
+      if (module_id == M[0].m_module_id) {
+        if (chi2 < M[0].m_chi2) {
+          M[0].m_chi2 = chi2;
+          M[0].m_hit_idx = hit_idx;
+        }
+      } else if (module_id == M[1].m_module_id) {
+        if (chi2 < M[1].m_chi2) {
+          M[1].m_chi2 = chi2;
+          M[1].m_hit_idx = hit_idx;
+        }
+      } else {
+        if (M[0].m_chi2 > M[1].m_chi2) {
+          if (chi2 < M[0].m_chi2) {
+            M[0] = {hit_idx, module_id, chi2};
+          }
+        } else {
+          if (chi2 < M[1].m_chi2) {
+            M[1] = {hit_idx, module_id, chi2};
+          }
+        }
+      }
+    }
+
+    HitMatch* find_overlap(int hit_idx, int module_id) {
+      if (module_id == M[0].m_module_id) {
+        if (M[1].m_hit_idx >= 0)
+          return &M[1];
+      } else if (module_id == M[1].m_module_id) {
+        if (M[0].m_hit_idx >= 0)
+          return &M[0];
+      } else {
+        if (M[0].m_chi2 <= M[1].m_chi2) {
+          if (M[0].m_hit_idx >= 0)
+            return &M[0];
+        } else {
+          if (M[1].m_hit_idx >= 0)
+            return &M[1];
+        }
+      }
+
+      return nullptr;
+    }
+  };
+
+  template <class T>
+  class CcPool {
+    std::vector<T> m_mem;
+    std::size_t m_pos = 0;
+    std::size_t m_size = 0;
+
+  public:
+    void reset(std::size_t size) {
+      if (size > m_mem.size())
+        m_mem.resize(size);
+      m_pos = 0;
+      m_size = size;
+
+      // printf("CcP reset to %zu\n", size);
```

Comment:
remove / protect with a debugging ifdef

Reply:
Addressed in Round 2 follow-up PR trackreco/cmssw#76 commit: https://github.com/trackreco/cmssw/pull/76/commits/d2c6a5ba4edbe357152710835beb41fb16faf51e